### DISCUSSION TITLE
HtmlToPdfConverter or pdf rendering using wkhtmltopdf error in serenity

### DISCUSSION THREAD

**Discussion by imhemayatsangin** [serenity-is/Serenity#6029] [Help] [answered]
Hello friends, I'm facing a problem when i want to generate the invoice which is in the Northwind demo in serenity .net core it gives me the following error:
`Value cannot be null. (Parameter 'UtilityExePath')'`
I have the wkhtmltopdf.exe file in the root directory of the project and also the path in the reportController.cs is as follows:
`   var wkhtmlPath = System.IO.Path.Combine(typeof(ReportController).Assembly.Location, "wkhtmltopdf.exe");`
Please guide me what is the issue and where im doing the mistake. i donwloaded the wkhtmltopdf.exe file from the internet but i dont have any idea where im making mistake.

Please help me out.

**Comment by minhhungit** [comment, marked as answer]
download file wkhtmltopdf.exe and add it into root folder

![image](https://user-images.githubusercontent.com/2279508/140041399-1b25b478-d7ae-4b72-80e0-b8f6585e33a8.png)

**Reply by imhemayatsangin** [reply]
As i have mentioned it in the issue that i have downloaded the file(wkhtmltopdf.exe) and it's already in the root directory and also it's copied to the bin directory as well. but still it's giving me error. do i need to add HtmlToPdfConverter.cs file anywhere to the project or not? because with whatever solutions posted here i tried but still getting the error. Please help me how to solve it?
